Source/DafnyLanguageServer/Workspace/NotificationPublisher.cs

```diff
@@ -1,16 +1,35 @@
-using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using Microsoft.Dafny.LanguageServer.Workspace.Notifications;
 using OmniSharp.Extensions.LanguageServer.Protocol.Document;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
 using OmniSharp.Extensions.LanguageServer.Protocol.Server;
 using System.Linq;
+using System.Net.Mime;
 using Microsoft.Extensions.Logging;
-using Serilog.Core;
 using Microsoft.Dafny.LanguageServer.Language;
+using Microsoft.Dafny.LanguageServer.Language.Symbols;
 using Microsoft.Extensions.Options;
+using OmniSharp.Extensions.LanguageServer.Protocol;
 
 namespace Microsoft.Dafny.LanguageServer.Workspace {
+  public record CompilationView(
+    DocumentTextBuffer TextDocumentItem,
+    IEnumerable<Diagnostic> ResolutionDiagnostics,
+    SymbolTable SymbolTable,
+    IReadOnlyDictionary<ImplementationId, ImplementationView> ImplementationViews,
+    bool ImplementationsWereUpdated,
+    IEnumerable<Diagnostic> GhostDiagnostics,
+    VerificationTree VerificationTree
+    ) {
+
+    public DocumentUri Uri => TextDocumentItem.Uri;
+    public int? Version => TextDocumentItem.Version;
+
+    public IEnumerable<Diagnostic> Diagnostics =>
+      ResolutionDiagnostics.Concat(ImplementationViews.Values.Select(v => v.Diagnostics).SelectMany(x => x));
```

Comment:
Probably for another PR, but with this architecture it seems much easier to fix the problem of resolution diagnostics lost in the middle of previous verification diagnostics. You could simply detect resolution diagnostics, and if so, change the level in implementation diagnostics to "Warning", right?

Reply:
Thanks, I plan to make that change after this PR.